rets/client/decoder.py

```diff
@@ -29,7 +29,10 @@ def decode(self, rows: Sequence[dict]) -> Sequence[dict]:
         def decode_field(field: str, value: str) -> Any:
             if value == '':
                 return None
-            return decoders[field](value)
+            try:
+                return decoders[field](value)
+            except Exception as e:
+                raise ValueError(f"Error decoding field {field} with value {value}. Error: {e}") from e
```

Comment:
You don't need the `Error: {e}` part, python would already show both the errors chained in the message with stack traces for you.

Reply:
I think the other error message is shown further up (like 30 lines up)
like 
```
ValueError: invalid literal for int() with base 10: '5.6'

The above exception was the direct cause of the following exception:
```

but I thought it'd be easier to have the error all in 1 place so that people don't have to realize they need to scroll up to find the "The above exception was the direct cause of the following exception:"

full stack trace using the current implementation:
```
...
{"severity":"INFO"     ,"time":"2021-05-24T21:48:14.024Z","logger":"dwellings.services.rets","event":"Updating User_OfficeName_Key to the datatype: Character",}
/Users/geobio/anaconda3/envs/dwellings/lib/python3.6/site-packages/rets/http/parsers/parse.py:120: FutureWarning: The behavior of this method will change in future versions. Use specific 'len(elem)' or 'elem is not None' test instead.
  comments=comments_elem and (comments_elem.text or None),
SystemMetadata(system_id='connectMLS', system_description='connectMLS RETS Server', system_date='2007-12-04T16:24:37', system_version='1.0.0', time_zone_offset=None, comments=<Element COMMENTS at 0x7fa7fc1bb248>)
Traceback (most recent call last):
  File "/Users/geobio/anaconda3/envs/dwellings/lib/python3.6/site-packages/rets/client/decoder.py", line 34, in decode_field
    return decoders[field](value)
ValueError: invalid literal for int() with base 10: '5.6'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "scripts/debug_raw_mls_query.py", line 63, in <module>
    debug_raw_mls_query()
  File "/Users/geobio/repos/dwellings/dwellings/factory.py", line 72, in use_app_wrapper
    return func(*args, **kwargs)
  File "/Users/geobio/anaconda3/envs/dwellings/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/geobio/anaconda3/envs/dwellings/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/geobio/anaconda3/envs/dwellings/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/geobio/anaconda3/envs/dwellings/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "scripts/debug_raw_mls_query.py", line 52, in debug_raw_mls_query
    fields=fields or get_fields_to_fetch(config),
  File "/Users/geobio/anaconda3/envs/dwellings/lib/python3.6/site-packages/rets/client/resource_class.py", line 66, in search
    rows = decoder.decode(result.data)
  File "/Users/geobio/anaconda3/envs/dwellings/lib/python3.6/site-packages/rets/client/decoder.py", line 39, in decode
    for row in rows)
  File "/Users/geobio/anaconda3/envs/dwellings/lib/python3.6/site-packages/rets/client/decoder.py", line 39, in <genexpr>
    for row in rows)
  File "/Users/geobio/anaconda3/envs/dwellings/lib/python3.6/site-packages/rets/client/decoder.py", line 38, in <genexpr>
    return tuple(OrderedDict((field, decode_field(field, value)) for field, value in row.items())
  File "/Users/geobio/anaconda3/envs/dwellings/lib/python3.6/site-packages/rets/client/decoder.py", line 36, in decode_field
    raise ValueError(f"Error decoding field {field} with value {value}. Error: {e}") from e
ValueError: Error decoding field ORM4_WID with value 5.6. Error: invalid literal for int() with base 10: '5.6'
```

Reply:
but if you prefer, I can delete it